inference-engine/src/gna_plugin/layers/gna_layer_info.hpp

```diff
@@ -201,13 +202,13 @@ class LayerInfo {
         return isOfType("concat");
     }
     bool isNonFunctional() const noexcept {
-        return isOfType("reshape") || isOfType("squeeze") || isOfType("unsqueeze");
+        return isOfType("reshape") || isOfType("squeeze") || isOfType("unsqueeze") || isTrivialPermute();
```

Comment:
this good - while you dont have tests case for it - we found it that split->trivial_permute->concat can reproduce issue with that

Reply:
@kmagiers, @dorloff  propose to file a ticket  that we need a test case for trivial permute detection - can someone do it?

Reply:
I believe that my test covers this case, meaning it will fail wihtout that. Let me verify that.

Reply:
ok, it is possible, but anyway it will make your case more fragile - any of 3 problems appeared only your test will fail every time - so more hard to debug.
1 - permute
2 - delayed memory execution prder
3 - addOutput (or lstmecell unroll)
4 - improved pointers arithmetics.

Reply:
Separate task was created to make tests. It will be separate PR